From 041f9386594d7b95dd80219c82a6cd3a9770fe30 Mon Sep 17 00:00:00 2001
From: Seth Jennings <sjenning@redhat.com>
Date: Mon, 18 Aug 2014 15:42:46 -0400
Subject: [edac] sb_edac: make minimal use of channel_mask

Message-id: <1408376571-8244-8-git-send-email-sjenning@redhat.com>
Patchwork-id: 87871
O-Subject: [RHEL7.1 PATCH 07/12] sb_edac: make minimal use of channel_mask
Bugzilla: 1127019
RH-Acked-by: Aristeu Rozanski <aris@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Aristeu Rozanski <aris@redhat.com>

Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Bugzilla: https://brewweb.devel.redhat.com/taskinfo?taskID=7802951
Upstream: d7c660b7dcb2f3c3835432cf733fba03507f4387

channel_mask will be used in the future to determine which group of memory
modules is causing the errors since when mirroring, lockstep and close page
are enabled you can't. While that doesn't happen, use the channel_mask to
determine the channel instead of relying on the MC event/exception.

Cc: Tony Luck <tony.luck@intel.com>
Signed-off-by: Aristeu Rozanski <aris@redhat.com>
Signed-off-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/sb_edac.c b/drivers/edac/sb_edac.c
index 117425a..ab12fb9 100644
--- a/drivers/edac/sb_edac.c
+++ b/drivers/edac/sb_edac.c
@@ -1678,6 +1678,9 @@ static void sbridge_mce_output_error(struct mem_ctl_info *mci,
   * EDAC core should be handling the channel mask, in order to point
   * to the group of dimm's where the error may be happening.
   */
+ if (!pvt->is_lockstep && !pvt->is_mirrored && !pvt->is_close_pg)
+  channel = first_channel;
+
  snprintf(msg, sizeof(msg),
    "%s%s area:%s err_code:%04x:%04x socket:%d channel_mask:%ld rank:%d",
    overflow ? " OVERFLOW" : "",
-- 
1.7.1